src/pc/protocols/usb_host_ep.cpp

```diff
@@ -0,0 +1,115 @@
+// project
+#include <cstdlib>
+#define MVLOG_UNIT_NAME xLinkUsb
+
+#include "XLink/XLinkLog.h"
+#include "XLink/XLinkPlatform.h"
+#include "XLink/XLinkPublicDefines.h"
+#include "usb_host_ep.h"
+#include "../PlatformDeviceFd.h"
+
+#if not defined(_WIN32)
+#include <unistd.h>
+#include <fcntl.h>
+#endif
+
+#include <stdlib.h>
+#include <cstring>
+
+static int usbFdRead, usbFdWrite;
+
+int usbEpInitialize() {
+    return 0;
+}
+
+int usbEpPlatformConnect(const char *devPathRead, const char *devPathWrite, void **fd)
+{
+    return X_LINK_ERROR;
+}
+
+int usbEpPlatformServer(const char *devPathRead, const char *devPathWrite, void **fd)
+{
+#if defined(_WIN32)
+    return X_LINK_ERROR;
+#else
+    char outPath[256];
+    strcpy(outPath, devPathWrite);
+    strcat(outPath, "/ep1");
+
+    char inPath[256];
+    strcpy(inPath, devPathWrite);
+    strcat(inPath, "/ep2");
+
+    int outfd = open(outPath, O_RDWR);
+    int infd = open(inPath, O_RDWR);
+
+    if(outfd < 0 || infd < 0) {
+	return -1;
+    }
+
+    usbFdRead = infd;
+    usbFdWrite = outfd;
+
+    *fd = createPlatformDeviceFdKey((void*) (uintptr_t) usbFdRead);
+
+    return 0;
+#endif
+}
+
+
+int usbEpPlatformClose(void *fdKey)
+{
+    int error;
+
+#if defined(_WIN32)
+    return X_LINK_ERROR;
+#else
+    if (usbFdRead != -1){
+	close(usbFdRead);
+	usbFdRead = -1;
+    }
+
+    if (usbFdWrite != -1){
+	close(usbFdWrite);
+	usbFdWrite = -1;
+    }
+#endif
```

Comment:
These should come from fdKeys while we are working on this. Remove global static usbFdRead/Write.

You'll have to tweak `static std::unordered_map<std::uintptr_t, void*> map;` to take a struct instead with two FDs, default api can return still just first FD, but add API to be able to query both FDs